server/events/vcs/mergeability.go

```diff
@@ -0,0 +1,129 @@
+package vcs
+
+import (
+	"github.com/google/go-github/v31/github"
+)
+
+// ValidStatusFilter implementations filter any valid statuses,
+// the definition of valid is up to the implementation.
+type ValidStatusFilter interface {
+	Filter(status []*github.RepoStatus) []*github.RepoStatus
```

Comment:
Maybe we should name this `FilterOut`? To me it sounds like we are filtering statuses to return matching values vs excluding matching values.

Reply:
Meh i think it makes sense because the struct is called ValidStatusFilter so we're filtering valid statuses.